frontend/awx/resources/templates/WorkflowVisualizer/Hooks/useWorkflowVisualizerToolbarActions.tsx

```diff
@@ -0,0 +1,39 @@
+import { useCallback } from 'react';
+import { useParams } from 'react-router-dom';
+import { Button, Toolbar, ToolbarContent, ToolbarItem } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { CloseIcon, PlusCircleIcon } from '@patternfly/react-icons';
+
+import { usePageNavigate } from '../../../../../../framework';
+import { AwxRoute } from '../../../../AwxRoutes';
+
+export function useWorkflowVisualizerToolbarActions() {
+  const { t } = useTranslation();
+  const pageNavigate = usePageNavigate();
+  const params = useParams<{ id: string }>();
+
+  const handleCancel = useCallback(() => {
+    pageNavigate(AwxRoute.WorkflowJobTemplateDetails, { params: { id: params.id } });
+  }, [pageNavigate, params.id]);
+
+  return (
+    <Toolbar style={{ flex: '1' }}>
+      <ToolbarContent>
```

Comment:
We don't need Toolbar and ToolbarContent, we could replace them with a react fragment. TopologyView's contextToolbar prop takes the value and nests it inside of a Toolbar, ToolbarContent, and ToolbarGroup. 

https://github.com/patternfly/react-topology/blob/869664948f24efa5b0299f5fa749f5bcfab8d444/packages/module/src/components/TopologyView/TopologyView.tsx#L111-L116